Installation issue: slepc

Our spack environment, based on E4S, picked up the latest version (`3.14.1`) of `slepc` with #20291.  This version did not build on ubuntu linux with `gcc@7.5.0`.  The concrete spec yaml file I attempted to install (attached) has all the specifics of the failed spec, but to summarize it was: `slepc@3.14.1 %gcc@7.5.0 arch=linux-ubuntu18.04-x86_64 /2m5avzr`.

Here's a summary from the error log:

```
/home/software/spack/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_pa/linux-ubuntu18.04-x86_64/gcc-7.5.0/python-3.8.6-szj7jukzoj34lkchuvzrzbh52kgesac7/bin/python3.8 /home/software/spack/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_path_placeholder/spack_pa/linux-ubuntu18.04-x86_64/gcc-7.5.0/petsc-3.14.1-dtldv5gbkfwz3672gw5uonhtexncgojo/share/petsc/examples/config/gmakegen.py --petsc-arch= --pkg-dir=/tmp/root/spack-stage/spack-stage-slepc-3.14.1-2m5avzreoqsbe57hcwryejaygpqssgae/spack-src --pkg-name=slepc --pkg-pkgs=sys,eps,svd,pep,nep,mfn,lme --pkg-arch=installed-arch-linux2-c-opt
gmakefile:72: recipe for target 'installed-arch-linux2-c-opt/lib/slepc/conf/files' failed
make[3]: *** [installed-arch-linux2-c-opt/lib/slepc/conf/files] Error 127
make[3]: *** Waiting for unfinished jobs....
/tmp/root/spack-stage/spack-stage-slepc-3.14.1-2m5avzreoqsbe57hcwryejaygpqssgae/spack-src/lib/slepc/conf/slepc_rules:121: recipe for target 'slepc_libs' failed
make[2]: *** [slepc_libs] Error 2
[1;31m*******************************ERROR************************************
  Error during compile, check installed-arch-linux2-c-opt/lib/slepc/conf/make.log
  Send all contents of ./installed-arch-linux2-c-opt/lib/slepc/conf to slepc-maint@upv.es
************************************************************************[0;39m[0;49m
makefile:30: recipe for target 'all' failed
make[1]: *** [all] Error 1
GNUmakefile:4: recipe for target 'all' failed
make: *** [all] Error 2
==> [2020-12-09-19:45:12.654425] Error: ProcessError: Command exited with status 2:
    'make' 'MAKE_NP=2'

4 errors found in build log:
     288    Using mpiexec: /home/software/spack/spack_path_placeholder/spack_pa
            th_placeholder/spack_path_placeholder/spack_path_placeholder/spack_
            path_placeholder/spack_path_placeholder/spack_path_placeholder/spac
            k_path_placeholder/spack_path_placeholder/spack_path_placeholder/sp
            ack_path_placeholder/spack_path_placeholder/spack_path_placeholder/
            spack_path_placeholder/spack_path_placeholder/spack_path_placeholde
            r/spack_path_placeholder/spack_path_placeholder/spack_path_placehol
            der/spack_path_placeholder/spack_path_placeholder/spack_pa/linux-ub
            untu18.04-x86_64/gcc-7.5.0/mpich-3.3.2-j3zl3i75xss7qzyp7mgqa27ugaj6
            3g3z/bin/mpiexec
     289    ------------------------------------------
     290    Using MAKEFLAGS: -j2 -l3.0  --no-print-directory -- SLEPC_DIR=/tmp/
            root/spack-stage/spack-stage-slepc-3.14.1-2m5avzreoqsbe57hcwryejayg
            pqssgae/spack-src PETSC_DIR=/home/software/spack/spack_path_placeho
            lder/spack_path_placeholder/spack_path_placeholder/spack_path_place
            holder/spack_path_placeholder/spack_path_placeholder/spack_path_pla
            ceholder/spack_path_placeholder/spack_path_placeholder/spack_path_p
            laceholder/spack_path_placeholder/spack_path_placeholder/spack_path
            _placeholder/spack_path_placeholder/spack_path_placeholder/spack_pa
            th_placeholder/spack_path_placeholder/spack_path_placeholder/spack_
            path_placeholder/spack_path_placeholder/spack_path_placeholder/spac
            k_pa/linux-ubuntu18.04-x86_64/gcc-7.5.0/petsc-3.14.1-dtldv5gbkfwz36
            72gw5uonhtexncgojo PETSC_ARCH=installed-arch-linux2-c-opt MAKE_NP=2
     291    ==========================================
     292    /home/software/spack/spack_path_placeholder/spack_path_placeholder/
            spack_path_placeholder/spack_path_placeholder/spack_path_placeholde
            r/spack_path_placeholder/spack_path_placeholder/spack_path_placehol
            der/spack_path_placeholder/spack_path_placeholder/spack_path_placeh
            older/spack_path_placeholder/spack_path_placeholder/spack_path_plac
            eholder/spack_path_placeholder/spack_path_placeholder/spack_path_pl
            aceholder/spack_path_placeholder/spack_path_placeholder/spack_path_
            placeholder/spack_path_placeholder/spack_pa/linux-ubuntu18.04-x86_6
            4/gcc-7.5.0/python-3.8.6-szj7jukzoj34lkchuvzrzbh52kgesac7/bin/pytho
            n3.8 /home/software/spack/spack_path_placeholder/spack_path_placeho
            lder/spack_path_placeholder/spack_path_placeholder/spack_path_place
            holder/spack_path_placeholder/spack_path_placeholder/spack_path_pla
            ceholder/spack_path_placeholder/spack_path_placeholder/spack_path_p
            laceholder/spack_path_placeholder/spack_path_placeholder/spack_path
            _placeholder/spack_path_placeholder/spack_path_placeholder/spack_pa
            th_placeholder/spack_path_placeholder/spack_path_placeholder/spack_
            path_placeholder/spack_path_placeholder/spack_pa/linux-ubuntu18.04-
            x86_64/gcc-7.5.0/petsc-3.14.1-dtldv5gbkfwz3672gw5uonhtexncgojo/shar
            e/petsc/examples/config/gmakegen.py --petsc-arch= --pkg-dir=/tmp/ro
            ot/spack-stage/spack-stage-slepc-3.14.1-2m5avzreoqsbe57hcwryejaygpq
            ssgae/spack-src --pkg-name=slepc --pkg-pkgs=sys,eps,svd,pep,nep,mfn
            ,lme --pkg-arch=installed-arch-linux2-c-opt
     293    gmakefile:72: recipe for target 'installed-arch-linux2-c-opt/lib/sl
            epc/conf/files' failed
  >> 294    make[3]: *** [installed-arch-linux2-c-opt/lib/slepc/conf/files] Err
            or 127
     295    make[3]: *** Waiting for unfinished jobs....
     296    /tmp/root/spack-stage/spack-stage-slepc-3.14.1-2m5avzreoqsbe57hcwry
            ejaygpqssgae/spack-src/lib/slepc/conf/slepc_rules:121: recipe for t
            arget 'slepc_libs' failed
  >> 297    make[2]: *** [slepc_libs] Error 2
     298    *******************************ERROR*******************************
            *****
     299      Error during compile, check installed-arch-linux2-c-opt/lib/slepc
            /conf/make.log
     300      Send all contents of ./installed-arch-linux2-c-opt/lib/slepc/conf
             to slepc-maint@upv.es
     301    *******************************************************************
            *****
     302    makefile:30: recipe for target 'all' failed
  >> 303    make[1]: *** [all] Error 1
     304    GNUmakefile:4: recipe for target 'all' failed
  >> 305    make: *** [all] Error 2
```

### Information on your system

Output of `spack debug report`:

```
$ spack debug report
* **Spack:** 0.16.0-202-9bcf99358
* **Python:** 3.6.9
* **Platform:** linux-ubuntu18.04-haswell
* **Concretizer:** original
```

The build failure happened inside of the following spack environment:

[spack.yaml.txt](https://github.com/spack/spack/files/5673905/spack.yaml.txt)

The concrete `spec.yaml` I attempted to install:

[slepc.yaml.txt](https://github.com/spack/spack/files/5673981/slepc.yaml.txt)

### Additional information

Spack build logs:

[spack-build-env.txt](https://github.com/spack/spack/files/5673918/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/5673919/spack-build-out.txt)

Other logs mentioned in the build log error messages (contents of `installed-arch-linux2-c-opt/lib/slepc/conf`) are attached below.

[slepcvariables.txt](https://github.com/spack/spack/files/5673946/slepcvariables.txt)
[slepcrules.txt](https://github.com/spack/spack/files/5673947/slepcrules.txt)
[make.log.txt](https://github.com/spack/spack/files/5673948/make.log.txt)
files.txt [empty]
[error.log.txt](https://github.com/spack/spack/files/5673956/error.log.txt)
[configure.log.txt](https://github.com/spack/spack/files/5673957/configure.log.txt)
[modules/slepc/3.14.1.txt](https://github.com/spack/spack/files/5673966/3.14.1.txt)

Mentioning the maintainers of `slepc`:

@balay @joseeroman

### General information

<!-- These boxes can be checked by replacing [ ] with [x] or by clicking them after submitting the issue. -->
- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/slepc/package.py
+++ b/var/spack/repos/builtin/packages/slepc/package.py
@@ -96,7 +96,7 @@ class Slepc(Package):
 
         if '+arpack' in spec:
             options.extend([
-                '--with-arpack-dir=%s' % spec['arpack-ng'].prefix.lib,
+                '--with-arpack-dir=%s' % spec['arpack-ng'].prefix,
             ])
             if spec.satisfies('@:3.12.99'):
                 arpackopt = '--with-arpack-flags'

